Packs/Jizo_NDR/Integrations/Jizo_NDR/Jizo_NDR.py

```diff
@@ -0,0 +1,516 @@
+            return response.json()
+        else:
+            raise DemistoException(response.text, response.status_code, response.reason)
+
+
+    def get_device_records(self,args:Dict[str,Any]):
+        """
+        Get jizo device records. You can filter by ip_src, mac, hostname.
+        One of this params is mandatory. You can filter also by timestamp or probe name
+
+        """
+        url = f'{self._base_url}/jizo_device_records'
+
+        response = requests.get(
+            url, params=args,headers=self._headers, verify=False
+        )
+        if response.status_code==200:
+            return response.json()
+        else:
+            raise DemistoException(response.text, response.status_code, response.reason)
+
+    def get_device_alerts(self,args:Dict[str,Any]):
+        """
+        Get jizo device alerts. You can filter by ip_src, ip_dest.
+        One of this params is mandatory. You can filter also by timestamp or probe name
+
+        """
+        url = f'{self._base_url}/jizo_get_devicealerts'
+
+        response = requests.get(
+            url, params=args,headers=self._headers, verify=False
+        )
+        if response.status_code==200:
+            return response.json()
+        else:
+            raise DemistoException(response.text, response.status_code, response.reason)
+
+    def get_alert_list(self, limit: int, start_time: str,last_id: int = 0) -> list[dict]:
+        """Call jizo_query_records endpoint and select specific fields
+        of response to return in alerts
+
+        Args:
+            limit (int): The number of items to generate.
+            start_time (str): first fetch time of incidents.
+            last_id(int): last incidents id
+
+        Returns:
+            list[dict]: data of formatted items
+        """
+
+        context_data= self.get_query_records(args={'datetime_from': start_time,'limit':limit})
+
+        alert_types = list(context_data.keys())
+
+        alerts: list[dict] = []
+        j=0
+        c=0
+        alert_data=context_data[alert_types[j]]['data']
+
+        for i in range(limit):
+            while i-c==len(alert_data) and j< len(alert_types)-1:
+                j+=1
+                c+= len(alert_data)
+                alert_data=context_data[alert_types[j]]['data']
+    
+            if bool(alert_data):
+                severity= alert_data[i-c].get('severity', '4')
+                category= alert_data[i-c].get('alert_category', '')
+                item = ITEM_TEMPLATE.format(id=last_id + i + 1,
+                                            alert_type=alert_types[j],
+                                            category=category,
+                                            severity=severity,
+                                            date= formatting_date(alert_data[i-c]['date']['date'])
+                                        )
+                dict_item = json.loads("{" + item + "}")
+                alerts.append(dict_item)
+
+        return alerts
```

Comment:
I will add comments in this function and change variable names.
The purpose of this function is to retrieve Alerts from the Jizo API. 
Jizo API returns 3 types of alerts (alerts flows, alerts files, alerts usecases). These alerts types represent the keys of the json response. Thus, the while loop is implemented to check if we reach the last element of alert data, or if it is empty.
For example: limit= 10 (we want to fetch 10 alerts). The response json is as following: 
{'alerts_flows':{ count: 0,data:[ ]} ,
'alerts_files': {count:2, data: [{ data1}, {data2}],
 'alerts_usecases': {count: 0, data: [ ]}
So the while loop will find that the data of the first alert type is empty, thus it will pass to the next alert type (j incrementation). I'm not sure if we could replace it with if statement. I will check. But, the purpose will be always the same. 
The variable c is like a counter here, to specify number of alerts that was retrieved.

I don't if it's clearer now or not. Anyway, I will add comments above code lines.

Reply:
Done

Reply:
In you example there are 2 alert_files and the rest is empty. If the limit=1, what would have happened? Will it fetch both data1 and data2? Will it fetch only data1 and in the next fetch retrieve data2? Will it retrieve only data1 and data2 won't be fetched?

Reply:
Please fix it, we don't want to miss any data in fetch-incidents.

Reply:
Maybe I didn't explain it properly. Each data represents an alert (data1 is the details of alert1, data2 is the details of alert2). So, if the limit(max_incidents_to_fetch)=1, we must only fetch one alert. Also, our API enables to retrieve the recent alerts, in fact it is updated in real time to fetch the recent alerts in first place.

Reply:
Ok, but we are looking at the fetch-incidents command. If limit(max_incidents_to_fetch)=1, than in the first minute we need to retrieve data1 and in the second minute we need to retrieve data2. Will it happen in the above code?

Reply:
No, I will add a function that checks if one of the last fetched alerts (retrieved in minute m-1) exists in list of alerts that we are retrieving in the minute m or not. If it is the case, we will pass to the next alert. I can also display the total number of existing alerts, so the user will have an idea and can fix limit according to that. Sounds good??

Reply:
I am not sure I understand, is there an option to sent the limit to the JizoM and get some kind of link or a flag that will tell us how to continue and retrieve the next alert in the next run of fetch-incidents?
Maybe we should schedule a call to discuss the available options.

Reply:
Yes, It will be better if we discuss this in a call. You can send me an invitation on my mail: 
souha.benhassine@sesame-it.fr